src/engine/runtime.js

```diff
@@ -2043,9 +2041,6 @@ class Runtime extends EventEmitter {
             this.profiler.start(stepProfilerId);
         }
 
-        // Clean up threads that were told to stop during or since the last step
-        this.threads = this.threads.filter(thread => !thread.isKilled);
-
```

Comment:
https://github.com/LLK/scratch-vm/pull/1683 is what makes this safe to remove, I think